src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/RabbitMQ/BasicGetIntegration.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ
+{
+    /// <summary>
+    /// RabbitMQ.Client BasicGet calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        Assembly = "RabbitMQ.Client",
+        Type = "RabbitMQ.Client.Impl.ModelBase",
+        Method = "BasicGet",
+        ReturnTypeName = "RabbitMQ.Client.BasicGetResult",
+        ParametersTypesNames = new[] { ClrNames.String, ClrNames.Bool },
+        MinimumVersion = "3.6.9",
+        MaximumVersion = "6.*.*",
+        IntegrationName = RabbitMQConstants.IntegrationName)]
+    public class BasicGetIntegration
+    {
+        private const string Command = "basic.get";
+
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.GetLogger(typeof(BasicGetIntegration));
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="queue">The queue name of the message</param>
+        /// <param name="autoAck">The original autoAck argument</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget>(TTarget instance, string queue, bool autoAck)
+        {
+            return new CallTargetState(scope: null, state: queue, startTime: DateTimeOffset.UtcNow);
```

Comment:
This is the one usage of the new `CallTargetState.StartTime` property. Since we must defer scope creation until the method returns (possible distributed parent context), we need to store the original start time somewhere, so I'm proposing we add it to the CallTargetState since this may not be the only case where we do this. I'd rather modify CallTargetState than mess with our Tracer's logic of initializing `SpanContext` objects

Reply:
It's ok for me to modify the CallTargetState struct with this new parameter, however we must be careful and only defer the creation of leaf spans in a distributed tracing, because any span created between OnMethodBegin and OnMethodEnd will have the wrong parent. I think it's ok in this case.

Reply:
Yes this is a leaf span, so this should be fine